include/proxsuite-nlp/bfgs-strategy.hpp

```diff
@@ -0,0 +1,115 @@
+/// @file
+/// @copyright Copyright (C) 2024 LAAS-CNRS, INRIA
+#pragma once
+
+#include "proxsuite-nlp/fwd.hpp"
+
+namespace proxsuite {
+namespace nlp {
+
+enum BfgsType { Hessian, InverseHessian };
+
+// forward declaration
+template <typename Scalar, BfgsType BFGS_TYPE> struct BfgsUpdateImpl;
+
+template <typename Scalar, BfgsType BFGS_TYPE = BfgsType::InverseHessian>
+class BfgsStrategy {
+  PROXSUITE_NLP_DYNAMIC_TYPEDEFS(Scalar);
+
+public:
+  BfgsStrategy(const int num_vars)
+      : M(Eigen::MatrixXd::Identity(num_vars, num_vars)), is_init(false),
```

Comment:
You should use the matrix type associated with the Scalar type

Reply:
good catch thanks